pkg/network/protocols/postgres/protocol.go

```diff
@@ -166,18 +177,39 @@ func (p *protocol) Stop(*manager.Manager) {
 	if p.eventsConsumer != nil {
 		p.eventsConsumer.Stop()
 	}
+	if p.pgKernelStopChannel != nil {
+		close(p.pgKernelStopChannel)
+	}
 }
 
 // DumpMaps dumps map contents for debugging.
 func (p *protocol) DumpMaps(w io.Writer, mapName string, currentMap *ebpf.Map) {
-	if mapName == InFlightMap { // maps/postgres_in_flight (BPF_MAP_TYPE_HASH), key ConnTuple, value EbpfTx
+	switch mapName {
+	case InFlightMap:
+		// maps/postgres_in_flight (BPF_MAP_TYPE_HASH), key ConnTuple, value EbpfTx
 		var key netebpf.ConnTuple
 		var value postgresebpf.EbpfTx
 		protocols.WriteMapDumpHeader(w, currentMap, mapName, key, value)
 		iter := currentMap.Iterate()
 		for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
 			spew.Fdump(w, key, value)
 		}
+	case KernelTelemetryMapPlain:
+		// postgres_kernel_msg_count_plain (BPF_ARRAY_MAP), key 0, value PostgresKernelMsgCount
+		var zeroKey uint32
+		var value postgresebpf.PostgresKernelMsgCount
+		protocols.WriteMapDumpHeader(w, currentMap, mapName, zeroKey, value)
+		if err := currentMap.Lookup(unsafe.Pointer(&zeroKey), unsafe.Pointer(&value)); err == nil {
+			spew.Fdump(w, zeroKey, value)
+		}
+	case KernelTelemetryMapTls:
+		// postgres_kernel_msg_count_tls (BPF_ARRAY_MAP), key 0, value PostgresKernelMsgCount
+		var zeroKey uint32
+		var value postgresebpf.PostgresKernelMsgCount
+		protocols.WriteMapDumpHeader(w, currentMap, mapName, zeroKey, value)
+		if err := currentMap.Lookup(unsafe.Pointer(&zeroKey), unsafe.Pointer(&value)); err == nil {
+			spew.Fdump(w, zeroKey, value)
+		}
```

Comment:
there is no diff between those cases, why not unifying them together?

```suggestion
	case KernelTelemetryMapPlain, KernelTelemetryMapTls:
		// postgres_kernel_msg_count_tls (BPF_ARRAY_MAP), key 0, value PostgresKernelMsgCount
		var zeroKey uint32
		var value postgresebpf.PostgresKernelMsgCount
		protocols.WriteMapDumpHeader(w, currentMap, mapName, zeroKey, value)
		if err := currentMap.Lookup(unsafe.Pointer(&zeroKey), unsafe.Pointer(&value)); err == nil {
			spew.Fdump(w, zeroKey, value)
		}
```